### PR TITLE
Lightweight fire sounds.

### PR DESCRIPTION
**Pull request by sofar**
As an alternative to removing sounds entirely, this plays sounds at
each location when an event happens. The sound plays for just a few
seconds, and then stops. While this does leave awkward silences,
it's perhaps better than no fire sounds at all.

The fire sound events happen when a new fire node is made, or when
the fire spread timers tick. This covers events where people ignite
something with flint and tinder as well.

### CODE REVIEW

**Comment by sofar** [comment]
@paramat perhaps this is a fair alternative?

**Comment by sofar** [comment]
Dropped the gain on all the samples down a lot - these are awkwardly loud.

**Comment by paramat** [comment]
Will try it.

File: mods/fire/init.lua [hunk 1]
```diff
@@ -29,6 +29,7 @@ minetest.register_node("fire:basic_flame", {
 	damage_per_second = 4,
 	groups = {igniter = 2, dig_immediate = 3, not_in_creative_inventory = 1},
 	on_timer = function(pos)
+		minetest.sound_play("fire_small", {pos = pos, gain = 0.2})
```
Comment (paramat) [inline mods/fire/init.lua]: If this is the last flame removed there will be sound for no flame.

File: mods/fire/init.lua [hunk 2]
```diff
@@ -41,11 +42,7 @@ minetest.register_node("fire:basic_flame", {
 
 	on_construct = function(pos)
 		minetest.get_node_timer(pos):start(math.random(30, 60))
-		minetest.after(0, fire.update_sounds_around, pos)
-	end,
-
-	on_destruct = function(pos)
-		minetest.after(0, fire.update_sounds_around, pos)
+		minetest.after(0, minetest.sound_play, "fire_small", {pos = pos, gain = 0.2})
```
Comment (paramat) [inline mods/fire/init.lua]: I think this should be in 'on construct' otherwise you get sound where no flame is.

**Comment by paramat** [comment]
Sorry but i think sounds starting and stopping, and playing for short periods, will actually seem more crude than no sounds at all.
These sounds will also interfere with ambience mods that provide fire sounds.
Lets leave it to optional mods (that i will provide) for sounds, for the short term.
:-1:

File: mods/fire/init.lua (hunk 2 above)
Comment (sofar) [inline mods/fire/init.lua]: the `on_destruct` is removed entirely - this code is in the on_construct handler instead.

File: mods/fire/init.lua [hunk 3]
```diff
@@ -272,6 +211,7 @@ else -- Fire enabled
 			if minetest.find_node_near(pos, 1, {"group:puts_out_fire"}) then
 				return
 			end
+			minetest.sound_play("fire_small", {pos = pos, gain = 0.2})
```
Comment (sofar) [inline mods/fire/init.lua]: No, it makes sure that a sound is played regularly at this fire node, irregardless of whether a new sound is played at a new place or not.

**Comment by sofar** [comment]
> Sorry but i think sounds starting and stopping, and playing for short periods followed by silence, will actually give a much worse impression than no sounds at all. 

Let other people please decide that as well, please. This isn't a bad alternative at all.

For instance, when you burn down an entire tree with this code, there's almost no pause in fire sounds - fire sounds continuously play while the tree burns down (at least in my tests they did, it's random of course, there could be a small gap here or there, but I didn't find it disturbing).

> These sounds will also interfere with ambience mods that provide fire sounds.

I don't think so. I lowered gain on them significantly, so if a mod adds additional sounds they are likely to only add to the ambiance.

File: mods/fire/init.lua (hunk 1 above)
Comment (sofar) [inline mods/fire/init.lua]: Sure, fixed that.

File: mods/fire/init.lua (hunk 2 above)
Comment (paramat) [inline mods/fire/init.lua]: Oops.

**Comment by paramat** [comment]
Fires are often small, especially when permanent flame is being used for campfires and fireplaces, having a sound play then stop will probably seem worse than no sound.
Nothing else in Minetest has ambience sounds so this will not seem inconsistent.

**Comment by paramat** [comment]
The ambience mods i am working on have very low gain at times, these sounds will interfere and spoil those mods. It's best this type of hacky code (mine and yours) is left to mods.

**Comment by paramat** [comment]
Actually the small fire sound almost comes to a clean stop already, just needs a short fade out right at the end.

**Comment by paramat** [comment]
Ok just tested by starting a forest fire, there is a problem. Because the ABM will add multiple flames at once the total sound gain becomes huge, even early on in a forest fire i had to stop the game because the sound became painfully loud. I also heard echo effects from sounds being started a fraction of a second apart.
But if the flame sound is made quieter it will then be too quiet for a small fire or single flame.

**Comment by paramat** [comment]
Tried reducing gains to 0.1 and removing line 214, still too loud and a weird sound from multiple sounds played at once.

**Comment by paramat** [comment]
Rejoining a world with a large forest fire i get warnings in terminal:
```
2016-12-06 05:09:35: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:35: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:35: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:35: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:35: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:35: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:35: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:35: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:35: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:35: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:35: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:35: WARNING[Main]: before createPlayingSound: invalid name
2016-12-06 05:09:35: WARNING[Main]: createPlayingSound: out of memory
2016-12-06 05:09:35: WARNING[Main]: before createPlayingSound: invalid name
2016-12-06 05:09:35: WARNING[Main]: updateListener: invalid name
2016-12-06 05:09:36: WARNING[Main]: before createPlayingSound: invalid name
2016-12-06 05:09:36: WARNING[Main]: createPlayingSound: out of memory
2016-12-06 05:09:36: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:36: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:36: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:36: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:36: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:36: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:36: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:36: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:36: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:36: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:36: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:36: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:36: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:36: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:36: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:36: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:36: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:36: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:36: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:36: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:36: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:36: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:36: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:36: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:36: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:36: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:36: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:36: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:36: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:36: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:36: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:36: WARNING[Main]: before createPlayingSound: invalid name
2016-12-06 05:09:36: WARNING[Main]: createPlayingSound: out of memory
2016-12-06 05:09:36: WARNING[Main]: before createPlayingSound: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSound: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:37: WARNING[Main]: before createPlayingSound: invalid name
2016-12-06 05:09:38: WARNING[Main]: updateListener: invalid name
2016-12-06 05:09:38: WARNING[Main]: before createPlayingSound: invalid name
2016-12-06 05:09:38: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:38: WARNING[Main]: before createPlayingSound: invalid name
2016-12-06 05:09:39: WARNING[Main]: before createPlayingSound: invalid name
2016-12-06 05:09:39: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:39: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:39: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:39: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:39: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:09:39: WARNING[Main]: before createPlayingSoundAt: invalid name
2016-12-06 05:10:00: WARNING[Main]: updateListener: invalid name
```

**Comment by paramat** [comment]
The large fire sound used for permanent flame is too loud, 0.1 seems about right.
0.1 for small fire sound is too quiet for a single flame, but far too loud in a forest fire.

How about i update my fire sounds PR to the simplest version, for your consideration?

**Comment by sofar** [comment]
```
2016-12-06 05:09:35: WARNING[Main]: createPlayingSoundAt: out of memory
2016-12-06 05:09:35: WARNING[Main]: before createPlayingSoundAt: invalid name
```

bwahaha, ok, that is definitely bad :)

**Comment by paramat** [comment]
Updated #1436 to a simpler version, with a sound setting too, as a possibility.

**Comment by sofar** [comment]
#1436 has none of the scale issues.

**Pull request closed without merging**